From 5e5cf12faf8d0070ccf379e5a1f4a52c72b31126 Mon Sep 17 00:00:00 2001
From: "Lain \"Fearyncess\" Yang" <i@lain.vg>
Date: Tue, 2 Jun 2026 17:22:07 +0800
Subject: [PATCH 823/823] AOSCOS: arm64: dts: qcom: radxa-dragon-q8b: set
 correct GPI DMA channel masks

Following fixes in 0e131c1c66cec, RADXA Q8B should has a similar fix in dts.

These DSDT info has been coming from extracted Q8B firmware packages[1].

[1]: https://dl.radxa.com/dragon/q8b/images/dragon-q8b_flat_build_wp_260523.zip

Fixes: "RADXA: arm64: dts: qcom: sc8280xp: Introduce Radxa Dragon Q8B"
Signed-off-by: Lain "Fearyncess" Yang <i@lain.vg>
---
 arch/arm64/boot/dts/qcom/sc8280xp-radxa-dragon-q8b.dts | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/arch/arm64/boot/dts/qcom/sc8280xp-radxa-dragon-q8b.dts b/arch/arm64/boot/dts/qcom/sc8280xp-radxa-dragon-q8b.dts
index d6d9a3ac3e068..c45fbdf025fe8 100644
--- a/arch/arm64/boot/dts/qcom/sc8280xp-radxa-dragon-q8b.dts
+++ b/arch/arm64/boot/dts/qcom/sc8280xp-radxa-dragon-q8b.dts
@@ -704,14 +704,17 @@ &dispcc0 {
 };
 
 &gpi_dma0 {
+	dma-channel-mask = <0x3>;
 	status = "okay";
 };
 
 &gpi_dma1 {
+	dma-channel-mask = <0x3>;
 	status = "okay";
 };
 
 &gpi_dma2 {
+	dma-channel-mask = <0x3>;
 	status = "okay";
 };
 
-- 
2.52.0

